### PR TITLE
[usm] Add debugging endpoint for listing USM traced programs

### PR DESCRIPTION
**Pull request by p-lambert**
### What does this PR do?

Add a debugging endpoint for the purposes of listing all traced USM programs and their PIDs.

### Motivation

Improve debugging workflow for uprobe-based programs.

### Additional Notes

Sample output:

``` 
[
  {
    "ProgramType": "shared_libraries",
    "FilePath": "/usr/lib/aarch64-linux-gnu/libcrypto.so.3",
    "PIDs": [
      759,
      36353,
      512,
      1386,
      2118,
      36350,
      42671,
      442,
      764,
      36354,
      36363,
      42667,
      45775,
      1,
      687,
      726,
      806
    ]
  },
  {
    "ProgramType": "shared_libraries",
    "FilePath": "/usr/lib/aarch64-linux-gnu/libgnutls.so.30.31.0",
    "PIDs": [
      2118,
      45775,
      726
    ]
  },
  {
    "ProgramType": "shared_libraries",
    "FilePath": "/lib/aarch64-linux-gnu/libssl.so.3",
    "PIDs": [
      45775
    ]
  }
]
```

### Reviewer's Checklist

- [ ] If known, an appropriate milestone has been selected; otherwise the `Triage` milestone is set.
- [ ] Use the `major_change` label if your change either has a major impact on the code base, is impacting multiple teams or is changing important well-established internals of the Agent. This label will be use during QA to make sure each team pay extra attention to the changed behavior. For any customer facing change use a releasenote.
- [ ] A [release note](https://github.com/DataDog/datadog-agent/blob/main/docs/dev/contributing.md#reno) has been added or the `changelog/no-changelog` label has been applied.
- [ ] Changed code has automated tests for its functionality.
- [ ] Adequate QA/testing plan information is provided if the `qa/skip-qa` label is not applied.
- [ ] At least one `team/..` label has been applied, indicating the team(s) that should QA this change.
- [ ] If applicable, docs team has been notified or [an issue has been opened on the documentation repo](https://github.com/DataDog/documentation/issues/new).
- [ ] If applicable, the `need-change/operator` and `need-change/helm` labels have been applied.
- [ ] If applicable, the `k8s/<min-version>` label, indicating the lowest Kubernetes version compatible with this feature.
- [ ] If applicable, the [config template](https://github.com/DataDog/datadog-agent/blob/main/pkg/config/config_template.yaml) has been updated.

### CODE REVIEW

File: cmd/system-probe/modules/network_tracer.go [hunk 1]
```diff
@@ -245,6 +246,7 @@ func (nt *networkTracer) Register(httpMux *module.Router) error {
 	})
 
 	httpMux.HandleFunc("/debug/usm_telemetry", telemetry.Handler)
+	httpMux.HandleFunc("/debug/usm_traced_programs", usm.TracedProgramsEndpoint)
```
Comment (guyarb) [inline cmd/system-probe/modules/network_tracer.go:249]: I think my endpoint name was better 😂

File: pkg/network/usm/utils/file_registry.go [hunk 2]
```diff
@@ -253,6 +260,13 @@ type registration struct {
 
 	// we are sharing the telemetry from FileRegistry
 	telemetry *registryTelemetry
+
+	// Note about the motivation for this field:
+	// a registration is tied to a PathIdentifier which is a global "identifier"
+	// to a file (dev, inode). Multiple file paths can point to the same
+	// underlying (dev, inode), so the `pathSample` here happens to be simply
+	// *one* of these file paths and use this mostly for debugging purposes
```
Comment (guyarb) [inline pkg/network/usm/utils/file_registry.go:268]: great comment

File: pkg/network/usm/utils/file_registry.go [hunk 3]
```diff
@@ -84,13 +84,19 @@ func NewFileRegistry(programName string) *FileRegistry {
...
 		procRoot:      kernel.ProcFSRoot(),
 		byID:          make(map[PathIdentifier]*registration),
 		byPID:         make(map[uint32]pathIdentifierSet),
 		blocklistByID: blocklistByID,
 		telemetry:     newRegistryTelemetry(programName),
 	}
+
+	// Add self to the debugger so we can inspect internal state of this
+	// FileRegistry using our debugging endpoint
+	debugger.Add(r)
```
Comment (guyarb) [inline pkg/network/usm/utils/file_registry.go:97]: when do we delete from the debugger?
the concern is having an ever increasing "in memory storage" for FileRegistry

Comment (guyarb) [reply pkg/network/usm/utils/file_registry.go:97]: wondering if we should have the `debugger` live all the time
a different approach would be - when triggering the endpoint, then iterate over all FileRegistries, and in the end, delete (the GC) the resources
that way we don't introduce a potential "leak"

Comment (p-lambert) [reply pkg/network/usm/utils/file_registry.go:97]: The debugger shouldn't do any long-lived memory allocations, so the memory footprint should be virtually zero.
The only thing it keeps throughout the whole lifecycle of system-probe is one pointer to each file registry.

**Review by guyarb — APPROVED** [review]
Looks good to me!
Please open a jira ticket + expand the pr description before merging

**Review by guyarb — APPROVED** [review]

**Review by hmahmood — APPROVED** [review]

File: pkg/network/usm/utils/ebpfoptions.go [hunk 4]
```diff
@@ -3,7 +3,7 @@
 // This product includes software developed at Datadog (https://www.datadoghq.com/).
 // Copyright 2016-present Datadog, Inc.
 
-//go:build linux_bpf
+//go:build linux
```
Comment (guyarb) [inline pkg/network/usm/utils/ebpfoptions.go:6]: I don't think we should change the build type of `ebpfoptions` to linux

**Pull request merged**